EngineLayer/Psm.cs

```diff
@@ -401,14 +401,14 @@ internal void Add(Psm psmParent)
         private Tuple<string, double?> Resolve(IEnumerable<double> enumerable)
         {
             double doubleTol = 1e-6;
-            var list = enumerable.ToList();
+            var list = enumerable.Distinct().ToList();
```

Comment:
that's Stefan's argument and I think it's a valid one; it's a difference of opinion on which is less confusing